client/state/notices/reducer.js

```diff
@@ -2,39 +2,103 @@
  * External dependencies
  */
 import { combineReducers } from 'redux';
+import omit from 'lodash/omit';
+import reduce from 'lodash/reduce';
+import { translate } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
 import {
-	NEW_NOTICE,
-	REMOVE_NOTICE,
-	ROUTE_SET,
-	SERIALIZE,
-	DESERIALIZE
+	NOTICE_CREATE,
+	NOTICE_REMOVE,
+	POST_SAVE_SUCCESS,
+	ROUTE_SET
 } from 'state/action-types';
+import { createReducer } from 'state/utils';
 
-export function items( state = [], action ) {
-	switch ( action.type ) {
-		case NEW_NOTICE:
-			return [ action.notice, ...state ];
-		case REMOVE_NOTICE:
-			return state.filter( ( notice ) => ( notice.noticeId !== action.noticeId ) );
-		case ROUTE_SET:
-			return state.filter( notice => {
-				const show = notice.isPersistent || notice.displayOnNextPage;
-				if ( notice.displayOnNextPage ) {
-					notice.displayOnNextPage = false;
+export const items = createReducer( {}, {
+	[ NOTICE_CREATE ]: ( state, action ) => {
+		const { notice } = action;
+		return {
+			...state,
+			[ notice.noticeId ]: notice
+		};
+	},
+	[ NOTICE_REMOVE ]: ( state, action ) => {
+		const { noticeId } = action;
+		if ( ! state.hasOwnProperty( noticeId ) ) {
+			return state;
+		}
+
+		return omit( state, noticeId );
+	},
+	[ ROUTE_SET ]: ( state ) => {
+		return reduce( state, ( memo, notice, noticeId ) => {
+			if ( ! notice.isPersistent && ! notice.displayOnNextPage ) {
+				return memo;
+			}
+
+			let nextNotice = notice;
+			if ( nextNotice.displayOnNextPage ) {
+				nextNotice = {
+					...nextNotice,
+					displayOnNextPage: false
+				};
+			}
+
+			memo[ noticeId ] = nextNotice;
+			return memo;
+		}, {} );
+	},
+	[ POST_SAVE_SUCCESS ]: ( state, action ) => {
+		const { post } = action;
+		if ( ! post.status ) {
+			return state;
+		}
+
+		const noticeId = action.type;
+
+		let count;
+		if ( state[ noticeId ] ) {
+			count = state[ noticeId ].count + 1;
+		} else {
+			count = 1;
+		}
+
+		let text;
+		switch ( post.status ) {
+			case 'trash':
+				if ( 1 === count ) {
+					text = translate( 'Post successfully moved to trash' );
+				} else {
+					text = translate(
+						'%d post successfully moved to trash',
```

Comment:
>This will never be used. Maybe we should update the eslint check for the `%d` to be optional in the singular case

Strange as it looks, the singular form here _will_ be used in some languages:

>Note that some languages use the singular form for other numbers (e.g. 21, 31 and so on [...])

https://codex.wordpress.org/I18n_for_WordPress_Developers#Plurals